### ISSUE TITLE
Relationship with time property between Nodes

### ISSUE THREAD

**Issue by daironpf** [jalakoo/mock-graph-data-generator-streamlit#16]
Hello, greetings

I'm generating a fake graph with python for a project I'm working on and I have to create millions of nodes, so I have **UserApp** nodes that have a name property: **dateTimeSignin** that stores the time the user signed in. Now I want to create a name relationship: **FRIEND_OF** that contains a property of type time and name: **fromThisMoment**
with the following feature:
`fromThisMoment = random time`
`(fromThisMoment > A.dateTimeSignin) and (fromThisMoment > B.dateTimeSignin)`

This type of property can be generated by the app you are creating here ? it's possible ?

Greetings and the app you are creating looks great and has a great use

**Comment by jalakoo** [comment]
Hi @daironpf!

I'm currently considering how best to add parameterization to the mock generator to support things like this.  For the time being you could do a little post processing to get the same effect.

So add the **FRIEND_OF** relationship with a datetime property (ie **createdAt**) using the datetime generator. After the data is generated and loaded to your database, use Cypher to update the **dateTimeSignin** properties of the **UserApp** nodes to match it's earliest **FRIEND_OF** relationship.

Something like:
```
MATCH (u:UserApp)-[r:FRIEND_OF]->(UserApp)
WITH u, max(datetime(r.createdAt).epochMillis) as earliestFriendOf
SET u.dateTimeSignin = datetime({epochMillis: earliesetFriendOf})
```

You'll probably need to edit that Cypher a bit since I'm writing it off the top of my head - but hopefully it gets the idea across.  I recommend generating a small dataset to try this out on first.

**Comment by daironpf** [comment]
Hi @jalakoo 

Currently I am still creating the fake graph for the company I am working for. I'm doing it based on ratios based on the number of users that they set in a config.

So they can scale the database to the size they want.

The time problem gave me a lot of work and I published it in the community, where they helped me greatly and here I leave the link in case it may be useful to you.

[Time problem solve link](https://community.neo4j.com/t/i-need-help-to-creating-a-random-date-as-a-property-of-a-relationship/63093)

I think your project is great and I wish you could take it in the future to the point of: 
1. writes to chatgpt the problem domain of your business together with the proportions for the relations and nodes; 
2. chatgpt returns the graph for arrow as you already have it, 
3. you adjust the names and details; then you skip all the data importer and import limits... 
4. you just run your data creation script based on your arrow data model and it generates the entire fake graph with millions or trillions of nodes.

Regards Dairon
[ThePandaCreator](https://www.youtube.com/@ElPandaCreador)

**Comment by jalakoo** [comment]
Thanks for the suggestions @daironpf!

Replies to each:
1. This one I could use some clarity with. You would like a graph with the proportions generated from a prompt, or to have a prompt generated graph that then could have proportions added?

2. Is your proposal different from the `ideate` tab's `Arrows.app compatible file` download button? It appears after a graph is generated from the prompt.  It's not the best transition I agree. I am in discussions with the arrows.app maintainer to find a better way of doing this.
<img width="821" alt="image" src="https://github.com/jalakoo/mock-graph-data-generator/assets/960717/f87e6be2-92b5-4c9b-b2d7-88aa4128f2cc">

3. Are you suggesting any edits to the arrows file would auto re-generate a new fake data set in a target neo4j database?

4. Yes! This is ultimately where I want to get this project to.

**Comment by jalakoo** [comment]
@daironpf - Latest updates I believe address items 2-4 from your earlier notes.  I'll take a look at the remaining suggestion for improving the GPT portion to help fill in the COUNT ranges the next time I do updates.  Likely will go with a GPT-Engineer type solution where an agent will ask some follow up questions to help define those numbers.

**Comment by daironpf** [comment]
@jalakoo Hello, I made a fake graph generator for the project I work on completely from scratch using Python, I made it in such a way that it is modular and generates millions of nodes in proportion to a root node, in my case as it is a social network it I do it based on the user. I made the time property a percentage of approach or distance to a time base property.

For example:
User:
name: "Pepe"
LocalDateTime signinMoment: "2014-05-30T08:21:38.459659000"

So when I create a Post in the fake graph
Post:
LocalDateTime postingTime = %time

Where the %time would be equal to a time that is in a range that goes from User.signinMoment(start) - Time.Established.End(end)(this can be the nowTime())

This is how it was done by setting time limits and then creating other relationships in equal proportions so that in the end everything will be perfect as a simulation of how a real user can act on a social network.

Greetings.
In a while I plan to try to make a more generic version for my fake graph generator and publish it on github for those who need it, but that still takes a time.
